src/vtquery.cpp

```diff
@@ -219,6 +236,70 @@ std::vector<vtzero::property> get_properties_vector(vtzero::feature& f) {
     return v;
 }
 
+double convert_to_double(value_type value) {
+    // float
+    if (value.which() == 1) {
+        return double(boost::get<float>(value));
+    }
+    // double
+    if (value.which() == 2) {
+        return boost::get<double>(value);
+    }
+    // int64_t
+    if (value.which() == 3) {
+        return double(boost::get<int64_t>(value));
+    }
+    // uint64_t
+    if (value.which() == 4) {
+        return double(boost::get<uint64_t>(value));
+    }
+    return 0.0f;
+}
+
+/// apply filters to a feature
+bool filter_feature(vtzero::feature& feature, std::vector<filter_struct> filters) {
+    using key_type = std::string;
+    using map_type = std::map<key_type, value_type>;
+    auto features_property_map = vtzero::create_properties_map<map_type>(feature);
+    bool reject_feature = false;
+    double epsilon = 0.001;
+    for (auto filter : filters) {
+        auto it = features_property_map.find(filter.key);
+        if (it != features_property_map.end()) {
+            value_type feature_value = it->second;
+            if (feature_value.which() == 5 && filter.value.which() == 5) {
+                bool feature_bool = boost::get<bool>(feature_value);
+                bool filter_bool = boost::get<bool>(filter.value);
+                if (filter.type == ne && feature_bool == filter_bool) {
+                    reject_feature = true;
+                } else if (filter.type == eq && feature_bool != filter_bool) {
+                    reject_feature = true;
+                }
+            } else if (feature_value.which() != 0 && feature_value.which() != 5 && filter.value.which() != 0 && filter.value.which() != 5) {
```

Comment:
Probably should be 1 <= .which() <= 4 to indicate numeric types instead of != 0 and != 5

Reply:
any reason why we can't apply these operations on uint (5) or sint (6) values?

Reply:
I was following [this](https://github.com/mapbox/vtzero/blob/master/doc/reading.md#creating-a-properties-map) and I think the uint and sint get mapped via the boost::variant defined as value_type to uint64_t and int64_t.
The `value_type = boost::variant<std::string, float, double, int64_t, uint64_t, bool>;` so the numeric ones are 1-5, unless I'm misunderstanding something.